### ISSUE TITLE
Cocoapods Targets Not Linking Dependencies Properly 

### ISSUE THREAD

**Issue by rvelasquez** [CocoaPods/CocoaPods#2849]
Hi,

I'm running into issues with missing symbols when trying to build my unit test target.  Specifically it can't find symbols found in the OCMockito project.  Here's my Podfile:

```ruby
platform :ios, '7.0'

pod 'TPKeyboardAvoiding'
pod 'UIView+FrameAdditions'
pod 'DejalActivityView'
pod 'AFNetworking'
pod 'UIAlertView+Blocks'
pod 'TDBadgedCell'
pod 'FLKAutoLayout'
pod 'Parse'
pod 'FlurrySDK'
pod 'TestFlightSDK'

target 'MyProjectUnitTestPods', :exclusive => true do
  pod 'OCMockito'
  link_with 'MyProjectUnitTests'
end
```

After running 'pod install' I can see that my unit test target (`MyProjectUnitTests`) is linked against `libPods.a` and `libPods-MyProjectUnitTestPods.a`.  I've also verified by looking in the Pods project that `libPods-MyProjectUnitTestPods.a` has a dependency on `Pods-MyProjectUnitTestPods-OCHamcrest` and `Pods-MyProjectUnitTestPods-OCMockito`.  However, when I try building `MyProjectUnitTests` it complains about not being able to find symbols that should be found in `Pods-MyProjectUnitTestPods-OCMockito`.  The only way I was able to get this target to build was to manually link the target with `Pods-MyProjectUnitTestPods-OCHamcrest` and `Pods-MyProjectUnitTestPods-OCMockito`.

So I think something is broken with how Cocoapods links libraries.

**Comment by kylef** [comment]
You'll notice when you `pod install` you get a warning:

```
[!] CocoaPods did not set the base configuration of your project because your project already has a custom config set. In order for CocoaPods integration to work at all, please either set the base configurations of the target `OCMockitoTestTests` to `Pods/Target Support Files/Pods-OCMockitoTestTestPods/Pods-OCMockitoTestTestPods.debug.xcconfig` or include the `Pods/Target Support Files/Pods-OCMockitoTestTestPods/Pods-OCMockitoTestTestPods.debug.xcconfig` in your build configuration.

[!] CocoaPods did not set the base configuration of your project because your project already has a custom config set. In order for CocoaPods integration to work at all, please either set the base configurations of the target `OCMockitoTestTests` to `Pods/Target Support Files/Pods-OCMockitoTestTestPods/Pods-OCMockitoTestTestPods.release.xcconfig` or include the `Pods/Target Support Files/Pods-OCMockitoTestTestPods/Pods-OCMockitoTestTestPods.release.xcconfig` in your build configuration.
```

tl;dr, this is because of how your Podfile is laid out. try the following Podfile:

```ruby
platform :ios, '7.0'

target 'OCMockitoTest' do
    pod 'TPKeyboardAvoiding'
    pod 'UIView+FrameAdditions'
    pod 'DejalActivityView' 
    pod 'AFNetworking'
    pod 'UIAlertView+Blocks'
    pod 'TDBadgedCell'
    pod 'FLKAutoLayout'
    pod 'Parse'
    pod 'FlurrySDK'
end

target 'OCMockitoTestTests' do
  pod 'OCMockito'
end
```

You will need to deintegrate the old integration beforehand else those warnings will persist:

```
$ gem install cocoapods-deintegrate
$ pod deintegrate
$ pod install
```

Now your project will build.
